src/binder/bind/bind_file_scan.cpp

```diff
@@ -75,6 +75,9 @@ std::unique_ptr<BoundBaseScanSource> Binder::bindScanSource(BaseScanSource* sour
         auto filePaths = bindFilePaths(fileSource->filePaths);
         // Bind file type.
         auto fileType = bindFileType(filePaths);
+        if (fileType != FileType::CSV && options.size() != 0) {
```

Comment:
I doubt this is the best place to perform the check because it's not scalable if we want to add another file type in the future.

A better design it to always parse options and save it in `ScanTableFuncBindInput`.  Then each scan function should have its own bind option logic. For csv scan, it will construct a csv option. For parquet it should validate no option. And I'm sure you also need to do it for npy.